packages/ui-library/src/components/date-time-picker/RuiDateTimePicker.spec.ts

```diff
@@ -734,4 +734,84 @@ describe('components/date-time-picker/RuiDateTimePicker.vue', () => {
   });
+
+  it('should allow typing leading zero for month (e.g., 08 for August)', async () => {
+    const date = dayjs('2022-01-01 09:09');
+    wrapper = createWrapper({
+      props: {
+        modelValue: date.toDate(),
+        type: 'date',
+      },
+    });
+    await vi.runOnlyPendingTimersAsync();
+
+    const inputField = wrapper.find('input');
+    await inputField.trigger('focus');
+    await vi.runOnlyPendingTimersAsync();
+
+    // Move to month field (skip day field)
+    await inputField.trigger('keydown.ArrowRight');
+    await vi.runOnlyPendingTimersAsync();
+
+    // Type "0" then "8" for August
+    await inputField.trigger('keydown', { code: 'Digit0', key: '0' });
+    await vi.runOnlyPendingTimersAsync();
+    await inputField.trigger('keydown', { code: 'Digit8', key: '8' });
+    await vi.runOnlyPendingTimersAsync();
+
+    const modelValue = wrapper.emitted('update:modelValue');
+    expect(modelValue).toBeTruthy();
+
+    const lastEmittedValue = modelValue?.at(-1)?.[0];
+    assert(lastEmittedValue && lastEmittedValue instanceof Date);
+
+    // Should be August (month 8)
+    expect(dayjs(lastEmittedValue).month()).toBe(7); // JavaScript months are 0-indexed (7 = August)
+    expect(dayjs(lastEmittedValue).format('YYYY-MM-DD')).toBe('2022-08-01');
+  });
+
+  it('should reject typing 0 alone for month', async () => {
+    const date = dayjs('2022-05-15 09:09');
+    wrapper = createWrapper({
+      props: {
+        modelValue: date.toDate(),
+        type: 'date',
+      },
+    });
+    await vi.runOnlyPendingTimersAsync();
+
+    const inputField = wrapper.find('input');
+    await inputField.trigger('focus');
+    await vi.runOnlyPendingTimersAsync();
+
+    // Move to month field (skip day field)
+    await inputField.trigger('keydown.ArrowRight');
+    await vi.runOnlyPendingTimersAsync();
+
+    // Clear the month field first
+    await inputField.trigger('keydown.Delete');
+    await vi.runOnlyPendingTimersAsync();
+
+    // Try to type just "0" for month
+    await inputField.trigger('keydown', { code: 'Digit0', key: '0' });
+    await vi.runOnlyPendingTimersAsync();
+
+    // Move to next field (should still be on month since 0 is not valid alone)
+    await inputField.trigger('keydown.ArrowRight');
+    await vi.runOnlyPendingTimersAsync();
+
+    const modelValue = wrapper.emitted('update:modelValue');
+
+    // Month should not have changed to 0 (which would become December of previous year)
+    // The model should either not have updated or should maintain the original month
+    if (modelValue) {
+      const lastEmittedValue = modelValue?.at(-1)?.[0];
+      if (lastEmittedValue instanceof Date) {
+        // Month should not be December (11) and year should not have changed
+        const emittedDate = dayjs(lastEmittedValue);
+        const shouldNotBe = dayjs('2021-12-15 09:09'); // Month 0 would cause this
+        expect(emittedDate.isSame(shouldNotBe)).toBe(false);
+      }
+    }
```

Comment:
instead of if here should we assert to make sure the correct value is always checked?

Is the value not always guaranteed to be set? 

We should at least expect the value to be truthy or assert that it's not 0/undefined